### PR TITLE
cleanup & dependency updates

### LINKED ISSUE #1: readme link should point to https://pulsar.apache.org/ not jetstream

### PR DESCRIPTION
**Pull request by MaxW1337**
Hey there, here i have a bunch of changes i've done for private projects which could may be relevant for you too. Will at least close #1 😄

### CODE REVIEW

File: pkg/pulsar/subscriber.go [hunk 1]
```diff
@@ -70,20 +65,25 @@ func NewSubscriberWithPulsarClient(conn pulsar.Client, logger watermill.LoggerAd
 	}, nil
 }
 
-// Subscribe subscribes messages from JetStream.
-func (s *Subscriber) Subscribe(ctx context.Context, topic string) (<-chan *message.Message, error) {
+// Subscribe subscribes messages from Pulsar.
+func (s *Subscriber) Subscribe(ctx context.Context, topic string, config SubscriptionConfig) (<-chan *message.Message, error) {
```
Comment (AlexCuse) [inline pkg/pulsar/subscriber.go]: This deviates from the watermill Subscriber interface and probably causes problems - ideally these options would be passed in to the NewSubscriber function as part of SubscriberConfig and used internally.

**Review by AlexCuse — CHANGES_REQUESTED** [review]
Thanks for waking this up, I threw it together in a couple hours one day and never looked again :sweat_smile: 

I have gotten more interested in pulsar as time goes on.  I remember running into problems trying to run the tests last time I tried but wonder if we could get this in shape to send back to ThreeDotsLabs as a supported pub/sub.

File: pkg/pulsar/subscriber.go (hunk 1 above)
Comment (AlexCuse) [reply pkg/pulsar/subscriber.go]: This is a nice trick you can use to make sure you stay in sync while working https://github.com/ThreeDotsLabs/watermill-nats/blob/master/pkg/jetstream/subscriber.go#L18

Comment (MaxW1337) [reply pkg/pulsar/subscriber.go]: Oh yeah, didn't thought of the Subscriber interface. Thanks for pointing that out and the trick used in jetstream subscriber.

**Review by AlexCuse — APPROVED** [review]

**Pull request merged**